xbmc/games/controllers/windows/GUIControllerWindow.cpp

```diff
@@ -280,6 +296,16 @@ void CGUIControllerWindow::OnFeatureSelected(unsigned int buttonIndex)
     m_featureList->OnSelect(buttonIndex);
 }
 
+void CGUIControllerWindow::UpdateControllerList(void)
+{
+  m_controllerList = new CGUIControllerList(this, m_featureList, m_gameClient);
```

Comment:
Could this cause a memory leak if UpdateControllerList() is called twice?

Reply:
Good point. Can you take a second look?

Reply:
I'm on it

Reply:
This function isn't safe to call from an add-on event handler. It must be called from the GUI thread. This is already done through the `GUI_MSG_REFRESH_LIST` message, but not for the correct events. See my fix: https://github.com/garbear/xbmc/commit/21341df79e28a78ffa16a0a6a2cf594bdedbe5f5

Reply:
Something seems wrong with the commit you linked